pkg/controllers/flannelmigration/migration_controller.go

```diff
@@ -365,8 +370,10 @@ func (c *flannelMigrationController) runIpamMigrationForNodes() ([]*v1.Node, err
 	for _, obj := range items {
 		node := obj.(*v1.Node)
 
-		val, _ := getNodeLabelValue(node, migrationNodeSelectorKey)
-		if val != "calico" {
+		migrationInProcess, _ := getNodeLabelValue(node, migrationNodeInProcessKey)
+		network, _ := getNodeLabelValue(node, migrationNodeSelectorKey)
+
+		if network != "calico" || migrationInProcess == "true" {
 			n := k8snode(node.Name)
 			if err := n.addNodeLabels(c.k8sClientset, nodeNetworkFlannel); err != nil {
```

Comment:
Why add flannel back if you've already changed network to calico?

Reply:
Yes. We do not need to. I though it would avoid another `if else`, but the downside of that is  Flannel pod having a chance of running on that node again. I will fix this.